NIFI-11201 Backported include-iotdb build profile

exceptionfactory opened the pull request:
# Summary

[NIFI-11201](https://issues.apache.org/jira/browse/NIFI-11201) Backports the `include-iotdb` Maven build profile addition.

# Tracking

Please complete the following tracking steps prior to pull request creation.

### Issue Tracking

- [X] [Apache NiFi Jira](https://issues.apache.org/jira/browse/NIFI) issue created

### Pull Request Tracking

- [X] Pull Request title starts with Apache NiFi Jira issue number, such as `NIFI-00000`
- [X] Pull Request commit message starts with Apache NiFi Jira issue number, as such `NIFI-00000`

### Pull Request Formatting

- [ ] Pull Request based on current revision of the `main` branch
- [X] Pull Request refers to a feature branch with one commit containing changes

# Verification

Please indicate the verification steps performed prior to pull request creation.

### Build

- [ ] Build completed using `mvn clean install -P contrib-check`
  - [ ] JDK 11
  - [ ] JDK 17

### Licensing

- [ ] New dependencies are compatible with the [Apache License 2.0](https://apache.org/licenses/LICENSE-2.0) according to the [License Policy](https://www.apache.org/legal/resolved.html)
- [ ] New dependencies are documented in applicable `LICENSE` and `NOTICE` files

### Documentation

- [ ] Documentation formatting appears as expected in rendered files

MikeThomsen reviewed (changes requested):
Bytecode incompatibility on Java 8:
<img width="1336" alt="image" src="https://user-images.githubusercontent.com/108184/220636150-d2c65e1d-603b-4825-94b0-571e19077910.png">

exceptionfactory commented:
Thanks @MikeThomsen, Java 8 build error is the result of incompatibilities in AspectJ 1.9.19 from [NIFI-11200](https://issues.apache.org/jira/browse/NIFI-11200). I reverted that upgrade on the support branch, and rebased this PR.

exceptionfactory commented:
@MikeThomsen after rebasing this PR from the current support branch, and also updating the `ci-workflow` to disable multi-threading for the compile step, the build completed on all platforms.
